### DISCUSSION TITLE
Dark Mode Switcher

### DISCUSSION THREAD

**Discussion by bnhr** [ben-rogerson/twin.macro#251] [Q&A] [answered]
I'm curious, but how do you guys create theme switcher using twin.macro? And also how to support tailwindcss/typography with dark mode?

**Comment by ben-rogerson** [comment, marked as answer]
Hey there, check out this resource on [how to create a theme switcher](https://github.com/ben-rogerson/twin.macro/blob/master/docs/advanced-theming.md).

Regarding @tailwindcss/typography, you'll need to [overwrite the color classes](https://github.com/tailwindlabs/tailwindcss-typography/blob/master/src/styles.js#L15) with css variables for your theme. [Check their readme for instructions](https://github.com/tailwindlabs/tailwindcss-typography#customization) on that.

**Reply by RasenGUY** [reply]
seems code-sandboxes here are not available anymore

**Reply by ben-rogerson** [reply]
They hadn't been updated in quite a while, I'd suggest taking a look online for some newer methods on how best to define a theme switcher with tailwind+[insert css-in-js-lib-here].

**Comment by irg1008** [comment]
Be aware this solution disables any background opacity in colors with css variables.

**Reply by ben-rogerson** [reply]
Here's how you add css opacity:

```js
// tailwind.config.js
module.exports = {
  theme: {
    colors: {
      primary: ({ opacityVariable, opacityValue }) => {
        if (opacityValue !== undefined) {
          return `rgba(var(--color-primary), ${opacityValue})`
        }
        if (opacityVariable !== undefined) {
          return `rgba(var(--color-primary), var(${opacityVariable}, 1))`
        }
        return `rgb(var(--color-primary))`
      },
    },
  },
  // ...
}
```

Then use rgb values for your colors:

```js
:root {
  --color-primary: 37, 99, 235;
  --color-secondary: 253, 224, 71;
}
```
